README.md

```diff
@@ -1,77 +1,141 @@
-`defaults/main.yml`) for the role should go here as these form an API of the
-role.
+### `mssql_accept_microsoft_odbc_driver_17_for_sql_server_eula`
 
-Variables that are not intended as input, like variables defined in
-`vars/main.yml`, variables that are read from other roles and/or the global
-scope (ie. hostvars, group vars, etc.) can be also mentioned here but keep in
-mind that as these are probably not part of the role API they may change during
-the lifetime.
+Set this variable to `true` to indicate that you accept EULA for installing the
+`msodbcsql17` package.
 
-Example of setting the variables:
+The license terms for this product can be downloaded
+from <https://aka.ms/odbc17eula> and found in `/usr/share/doc/msodbcsql17/LICENSE.txt`.
 
-```yaml
-template_foo: "oof"
-template_bar: "baz"
-```
+Default: `false`
 
-### Variables Exported by the Role
+### `mssql_accept_microsoft_cli_utilities_for_sql_server_eula`
 
-This section is optional.  Some roles may export variables for playbooks to
-use later.  These are analogous to "return values" in Ansible modules.  For
-example, if a role performs some action that will require a system reboot, but
-the user wants to defer the reboot, the role might set a variable like
-`template_reboot_needed: true` that the playbook can use to reboot at a more
-convenient time.
+Set this variable to `true` to indicate that you accept EULA for installing the
+`mssql-tools` package.
 
-Example:
+The license terms for this product can be downloaded
+from <http://go.microsoft.com/fwlink/?LinkId=746949> and found in
+`/usr/share/doc/mssql-tools/LICENSE.txt`.
 
-`template_reboot_needed` - default `false` - if `true`, this means
-a reboot is needed to apply the changes made by the role
+Default: `false`
 
-## Dependencies
+### `mssql_accept_microsoft_sql_server_2019_standard_eula`
 
-A list of other roles hosted on Galaxy should go here, plus any details in
-regards to parameters that may need to be set for other roles, or variables
-that are used from other roles.
+Set this variable to `true` to indicate that you accept EULA for using the
+`mssql-conf` utility.
 
-## Example Playbook
+The license terms for this product can be found in `/usr/share/doc/mssql-server`
+or downloaded from <https://go.microsoft.com/fwlink/?LinkId=2104078&clcid=0x409>.
+The privacy statement can be viewed at
+<https://go.microsoft.com/fwlink/?LinkId=853010&clcid=0x409>.
+
+Default: `false`
+
+### `mssql_password`
+
+The password for the database sa user. The password must have a minimum length
+of 8 characters, include uppercase and lowercase letters, base 10 digits or
+non-alphanumeric symbols. Do not use single quotes ('), double quotes ("), and
+spaces in the password because `sqlcmd` cannot authorize when the password
+includes those symbols.
+
+When running this role on a host that has MSSQL set up, the mssql_password
+variable overwrites the existing sa user password to the one that you specified.
+
+Default: `None`
```

Comment:
do you mean the literal string `None`?

Reply:
No, I mean that this variable is not defined by default

Reply:
Usually we define all documented input variables in defaults, and those that do not have a good default are set to the YAML/JSON special `null` value. You can then discover what parameters are supported by loading the role defaults (roles do not currently have a better way of introspection and Foreman/Satellite relies on this).

Reply:
So you suggest defining all vars that users can input with the role in defaults.yml. This sounds like a good idea, and indeed other services will be able to read acceptable variables much easier this was, and in an automatic manner, not just reading the README.md.
I often use the check `mssql_password is defined` to evaluate if a user wants to change the password or not. How to do the same when `mssql_password = null`?
One thing that worked for me is checking `mssql_password == None`. This evaluates to `true` when `mssql_password = null`. Is that the way to go?

Reply:
Another question @pcahyna, did you mean by the first comment that README should mark default as `null`, not `None`?

Reply:
> So you suggest defining all vars that users can input with the role in defaults.yml. This sounds like a good idea, and indeed other services will be able to read acceptable variables much easier this was, and in an automatic manner, not just reading the README.md.
> I often use the check `mssql_password is defined` to evaluate if a user wants to change the password or not. How to do the same when `mssql_password = null`?
> One thing that worked for me is checking `mssql_password == None`. This evaluates to `true` when `mssql_password = null`. Is that the way to go?

https://jinja.palletsprojects.com/en/2.9.x/templates/#none

In Jinja2 the literal is `none`, with `None` also accepted. In YAML, the literal is `null`. In Python, it is `None`.  https://jinja.palletsprojects.com/en/2.9.x/templates/#literals

But by using the `none` test, you won't have to compare to the literal value.

Reply:
Thank you. I've implemented this in the last commit.

Reply:
> Another question @pcahyna, did you mean by the first comment that README should mark default as `null`, not `None`?

Certainly it should not say that the default is `None`, because it looks like a literal string `None`, which is certainly incorrect. The default is not the string `None`. You may say that there is no default, i. e. default: none (without the quotes) (which is not 100% correct, because there is a value in defaults), or say what is the default behavior without specifying the actual default value (which is what the selinux role does).

Reply:
In README, I changed the default to `null`, it is the same in the documentation of modules. For example, in file and service module `Default: (null)`. I think this is fair.

Reply:
We also need to document that the `mssql_password` parameter is required in certain cases - initial setup, and anytime you specify a sql file.